src/devices/video/mc6847.cpp

```diff
@@ -76,25 +76,42 @@
 #include "emu.h"
 #include "video/mc6847.h"
 
-
 //**************************************************************************
 //  CONSTANTS
 //**************************************************************************
 
-#define TOP_BORDER              25
-#define USE_HORIZONTAL_CLIP     false
+#define LINES_TOP_BORDER				(25)
+#define LINES_ACTIVE_VIDEO				(192)
+#define LINES_BOTTOM_BORDER				(26)
+#define LINES_VERTICAL_RETRACE			(6)
+#define LINES_UNTIL_RETRACE				(LINES_TOP_BORDER + LINES_ACTIVE_VIDEO + LINES_BOTTOM_BORDER)
+#define LINES_UNTIL_VBLANK				(LINES_UNTIL_RETRACE + LINES_VERTICAL_RETRACE)
+
+#define USE_HORIZONTAL_CLIP				false
+
+#define CLOCKS_HSYNC_PERIOD				(228)
+#define	CLOCKS_FRONT_PORCH_DURATION		(7)
+#define CLOCKS_ACTIVE_VIDEO				(128)
+#define CLOCKS_L_OR_R_BORDER			(29)
+#define CLOCKS_ACTIVE_VIDEO_AND_BORDERS	(CLOCKS_ACTIVE_VIDEO + (2*CLOCKS_L_OR_R_BORDER))
+#define TIMER_HSYNC_OFF_TIME			(CLOCKS_ACTIVE_VIDEO_AND_BORDERS + CLOCKS_FRONT_PORCH_DURATION)
+#define TIMER_HSYNC_ON_TIME				(TIMER_HSYNC_OFF_TIME + 16.5)
+
+// FSYNC occurs just after active video, just before right border
+#define TIMER_FSYNC_TIME				(CLOCKS_ACTIVE_VIDEO + CLOCKS_L_OR_R_BORDER)
+
+// These units are the screen device's bitmap pixels.  Multiplied by 2 because
+// the pixel clock is the mc6847's clock * 2
+#define BMP_L_OR_R_BORDER				(CLOCKS_L_OR_R_BORDER * 2)
+#define BMP_ACTIVE_VIDEO				(CLOCKS_ACTIVE_VIDEO * 2)
 
-#define TIMER_HSYNC_PERIOD      (228)
-#define TIMER_HSYNC_OFF_TIME    (10.0)
-#define TIMER_HSYNC_ON_TIME     (TIMER_HSYNC_OFF_TIME + 16.5)
-#define TIMER_FSYNC_OFF_TIME    (TIMER_HSYNC_PERIOD * TOP_BORDER + TIMER_HSYNC_ON_TIME)
-#define TIMER_FSYNC_ON_TIME     (TIMER_HSYNC_PERIOD * (TOP_BORDER + 192) + TIMER_HSYNC_ON_TIME)
 
 #define LOG_SCANLINE (1U << 1)
 #define LOG_HSYNC    (1U << 2)
 #define LOG_FSYNC    (1U << 3)
 #define LOG_FLUSH    (1U << 4)
 #define LOG_INPUT    (1U << 5)
+#define LOG_NEXTLINE (1U << 6)
 #define VERBOSE (0)
 #include "logmacro.h"
```

Comment:
This block should really be just after the header declarations, as conventionally done by basically any device/driver that uses `logmacro.h`

Reply:
Done.